### PR TITLE
Memcache / Memcached: everything is broken

### PR DESCRIPTION
**Pull request by bryanlatten**
Both Memcache and Memcached have completely borked processes and status between 5.6 and 7.0. Fixing these currently broken aspects before moving onto a Redis adapter.

- Memcached: fixed for 2.x.x + 3.x.x breaking signatures 
- Memcache: fix for broken signature between PHP 5.6 + PHP 7.0 versions of same release (!)
- Tests: updated to latest phpunit with breaking/warning test changes 
- Fixed broken tests, when possible

Will need to unfortunately merge over failed tests on travis. Due to this bug: https://github.com/php-memcached-dev/php-memcached/issues/126

But here's the local output from compatible versions: 
screen shot 2016-10-18 at 3 43 28 pm

### CODE REVIEW

File: src/Adapters/MemcacheAdapter.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 use Behance\NBD\Cache\AdapterAbstract;
 use Behance\NBD\Cache\Exceptions\SystemRequirementException;
+use ReflectionMethod;
```
Comment (jimdelois) [inline src/Adapters/MemcacheAdapter.php]: Why not just write `\ReflectionMethod` below and remove this?

File: src/Adapters/MemcachedAdapter.php [hunk 2]
```diff
@@ -74,7 +86,9 @@ protected function _getMulti( array $keys ) {
 
     $null = null;
 
-    return $this->_connection->getMulti( $keys, $null, \Memcached::GET_PRESERVE_ORDER );
+    return ( $this->_memcached_version_3_0 )
+           ? $this->_connection->getMulti( $keys, \Memcached::GET_PRESERVE_ORDER )
+           : $this->_connection->getMulti( $keys, $null, \Memcached::GET_PRESERVE_ORDER );
```
Comment (jimdelois) [inline src/Adapters/MemcachedAdapter.php]: This is, quite literally, _unbelievable_.

**Comment by jimdelois** [comment]
👍

**Comment by be-dmitry** [comment]
👍

**Pull request merged**